app/javascript/components/Editor.tsx

```diff
@@ -30,6 +30,8 @@ import { RunTestsButton } from './editor/RunTestsButton'
 import { SubmitButton } from './editor/SubmitButton'
 import { useIsMounted } from 'use-is-mounted'
 import { camelizeKeys } from 'humps'
+import { useSaveFiles } from './editor/file-editor/useSaveFiles'
+import { isEqual } from 'lodash'
```

Comment:
We need this to perform deep equality between the current `files` array and the initial `files` array passed.